creator-node/src/services/stateMachineManager/stateReconciliation/updateReplicaSet.jobProcessor.ts

```diff
@@ -362,25 +363,31 @@ const _selectRandomReplicaSetNodes = async (
   ) {
     const randomHealthyNode = _.sample(healthyNodes)
 
-    // If node is already present in new replica set or is part of the exiting replica set, keep finding a unique healthy node
+    // If node is already present in new replica set or is part of the existing replica set, keep finding a unique healthy node
     if (
       newReplicaNodesSet.has(randomHealthyNode) ||
       healthyReplicaSet.has(randomHealthyNode)
     )
       continue
 
+    // If the node was marked as healthy before, keep finding a unique healthy node
+    if (unhealthyReplicasSet.has(randomHealthyNode)) {
+      logger.warn(
+        `Selected node ${randomHealthyNode} that is marked as healthy now but was previously marked as unhealthy. Unselecting it and finding another healthy node...`
+      )
+      continue
+    }
+
     // Check to make sure that the newly selected secondary does not have existing user state
     try {
       const clockValue = await retrieveClockValueForUserFromReplica(
         randomHealthyNode,
         wallet
       )
+      newReplicaNodesSet.add(randomHealthyNode)
       if (clockValue === -1) {
```

Comment:
i think this condition needs to change to !== -1 to make the log accurate. if clock === -1 there was no prior state

Reply:
whoops, fixed that typo